### ISSUE TITLE
mariadb : bug in limit

### ISSUE THREAD

**Issue by dominiquefournier** [PanCakeConnaisseur/user_backend_sql_raw#4]
MySQL/MariaDB don't supports the limit prepare in PARAM_STR. It must be a PARAM_INT.

I propose the following patch for lib/UserBackend.php

```
107c107
< 			$parameterSubstitutions['limit'] = intval ($limit);
---
> 			$parameterSubstitutions['limit'] = $limit;
114c114
< 			$parameterSubstitutions['offset'] = intval ($offset);
---
> 			$parameterSubstitutions['offset'] = $offset;
122,129c122
<     foreach ($parameterSubstitutions as $param => $val)
<     {
<       if (is_int ($val))
<         $statement->bindValue(":".$param, $val, \PDO::PARAM_INT);
<       else
<         $statement->bindValue(":".$param, $val, \PDO::PARAM_STR);
<     }
< 		$statement->execute();
---
> 		$statement->execute($parameterSubstitutions);
```

**Comment by PanCakeConnaisseur** [comment]
@dominiquefournier: Thank you very much for this report. Could please elaborate on what exactly is the problem and provide error messages? I would really like to avoid having any casts in the code because they make code harder to understand and therefore less maintainable.

Also I think your patch is reversed (from your change to the original version). Can you please provide a git style diff or make a pull request?

**Comment by PanCakeConnaisseur** [comment]
@dominiquefournier : Thank you very much for the report.

1. Is the else branch necessary, i.e. setting the parameter to string explicitly?
2. Your patch seems to be reversed, i.e. from your version to the original. Could you provide a git style patch in the other direction, please?
3. Also I think this should go in a separate method, to not  pollute the original method with workarounds specific to mariadb.

**Comment by dominiquefournier** [comment]
Hi !

Point 1. Yes, the "else" is needed as only the limit will be in integer. 
The other ones (like the username) has to be in string.

Point 2. Of course, I will, but I am sorry, I don't know how I can do 
that. Do you have a link or could you explain me how I should do ?

Point 3. If you want. Do you have an idea how you would like to separate 
the method ?



Le 17/07/2018 à 04:29, PanCakeConnaisseur a écrit :

**Comment by tisoft** [comment]
I have created a pull request with the patch from @dominiquefournier applied to it for your convenience.

I'm using this code now myself for connecting my nextcloud to my mail database. Thanks @PanCakeConnaisseur for the addon and @dominiquefournier for the mariadb fix!

**Comment by PanCakeConnaisseur** [comment]
Should be resolved by fd5270135d66f8ac6290e10f9001c88567207732 which is included in the v1.0.5 release.
